docs/architecture.md

```diff
@@ -190,3 +320,13 @@ await client.query(Q1) // returns { _id: 'my-todo', _type: 'io.cozy.todos', labe
 
 Q1 retrieves the full content of 'my-todo' while Q2 only retrieves its label. When Q2 is run, 'my-todo' already exists in the store, as Q1 was previsouly run. However, the fields retrieved by Q1 for 'my-todos' but absent from Q2 are not lost. This way, when Q1 is run again, the results fetched from the store (thanks to the fetch policy) will include all the fields.
 
+The problem is when some fields are removed from a document: the store will not reflect this change and result in stale data.
```

Comment:
Maybe we should add a warning emoji for that? And also say, that to avoid this, instead of removing a field, you can set it to undefined as a workaround?